UnitySetup/UnitySetup.psm1

```diff
@@ -475,7 +475,17 @@ function Install-UnitySetupInstance {
                 $localDestinations += , $Destination
             }
             else {
-                $localDestinations += , "C:\Program Files\Unity-$($i.Version)"
+                switch (Get-OperatingSystem) {
+                    ([OperatingSystem]::Windows) {
+                        $localDestinations += , "C:\Program Files\Unity\Hub\Editor\$($i.Version)"
```

Comment:
Different users may have chosen different Hub install location by clicking the Gear icon in the hub. For example, I install to my D: drive because my C: drive is an SSD with limited space. Can we reverse-engineer how this setting is saved and use the actual Hub setting instead of hard-coding this path?

Reply:
Unsure where exactly this value is saved.

Mainly I wanted to update this so that I can utilize it in Azure DevOps CI/CD pipelines where it's generally a clean install each time. One of the tasks we have looks for the default unity installation path.

Reply:
A manual work-around is in my PR which adds `BasePath` as an optional parameter which supports installs on other drives while also retaining `$($i.Version)` in the path. A bit more flexible than the current `Destination` parameter.

The install path might be located in the registry. Not sure where something like that would be on macOS. Would recommend to put that behind `Get-OperatingSystem` check if it does exist.

Reply:
I think it makes sense to use the default install path with a manual override if needed.

Reply:
It looks like this is saved in c:\users\YOU\AppData\Roaming\UnityHub\secondaryInstallPath.json. 

I haven't investigated thoroughly, but I think it may be empty by default and then is filled in if you click the gear, set a path, and exit the Hub. Perhaps we could create a GetHubInstallPath helper function that returns the default directory, overridden by the secondaryInstallPath.json (if non-empty), gated by a Get-OperatingSystem check?

Would you be willing to use this setting? Or should I open up an issue saying that UnitySetup should react to the Hub's install path?

Reply:
Awesome investigation skills!!! Yeah we should def check this location first and then use the default path if the user didn't supply a custom path.

Unfortunately, I'm not sure I have time to do this. Is this something we can do as a post PR step?

Reply:
Sure. Go ahead and complete this PR and then please open an issue saying that we should check the Hub setting.